### ISSUE TITLE
0.0.79 regression: ParamSpec forwarding gives a callback the type of an earlier argument

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ty#4493]
### Summary

This example passes with ty 0.0.78 and fails with ty 0.0.79 (Python version 3.12).

```python
from typing import Any, Callable

def capybara[**P](fn: Callable[P, Any], /, *args: P.args, **kwargs: P.kwargs): ...

def hutia(name: str, fn: Callable[[int], Any]): ...

capybara(hutia, "hay", lambda animal: animal)
```

Expected: the callback parameter `animal` is `int`, as declared by `hutia`.

Actual in 0.0.79:

```text
callback_minimal.py:7:24: error[invalid-argument-type] Argument to function `capybara` is incorrect: Expected `(int, /) -> Any`, found `(animal: str) -> str`
Found 1 diagnostic
```

This was reduced from a call to `asyncio.to_thread` with a callback argument. The earlier `str` parameter appears to affect inference of the nested callback. No asyncio dependency is needed for the reduced case.

### Version

0.0.79
